### PR TITLE
Feature/singup

### PR DESCRIPTION
**Pull request by rams23**
Add signup feature (#10).

Missing things:
* make roles and devops dynamic (#32)
* add error translations in form
* set user logged
* start email verification flow (#12)
* check on scam emails(?)
* security rules on firestore

### CODE REVIEW

File: packages/game-app/src/_shared/index.ts [hunk 1]
```diff
@@ -2,9 +2,13 @@ import { buildStore } from './store';
 import CONFIG from '@pipeline/app-config';
 import firebase from 'firebase/app';
 import 'firebase/analytics';
+import 'firebase/firestore';
+import 'firebase/functions';
```
Comment (Alb93) [inline packages/game-app/src/_shared/index.ts:6]: why?

File: packages/game-app/src/signup/apis/executeSignup.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      });
+      return {
+        id: user.uid,
+        email: user.email!,
+      };
+    } else {
+      throw new Error('user-not-created');
+    }
+  } catch (e) {
+    try {
```
Comment (Alb93) [inline packages/game-app/src/signup/apis/executeSignup.ts:26]: TODO: add custom firebase errors handling

File: packages/game-app/cypress/integration/signup.spec.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    cy.get('body').should('contain', 'signup.invalidEmail')
+  });
+
+  it("should show invalid password error", () => {
+    cy.getInputByName('password').type('123456789');
+    cy.get('button').click();
+    cy.get('body').should('contain', 'signup.passwordRequirements')
+  });
+
+  it("should show not contains invalid password error", () => {
```
Comment (GabrieleMazzola) [inline packages/game-app/cypress/integration/signup.spec.ts:26]: I would rephrase this with "should not show invalid password error for correct password"

File: packages/game-app/src/_shared/index.ts [hunk 4]
```diff
@@ -18,6 +22,13 @@ export function bootstrap() {
     messagingSenderId: CONFIG.REACT_APP_FIREBASE_CONFIG_MESSAGING_SENDER_ID,
     appId: CONFIG.REACT_APP_FIREBASE_CONFIG_APP_ID,
   });
+  if (config.REACT_APP_FIREBASE_USE_EMULATORS === 'true') {
+    firebase.firestore().settings({ experimentalForceLongPolling: true });
+    firebase.auth().useEmulator('http://localhost:9099/');
```
Comment (GabrieleMazzola) [inline packages/game-app/src/_shared/index.ts:27]: Why is this line using the full URL while the others don't?

File: packages/game-app/src/_shared/index.ts (hunk 1 above)
Comment (GabrieleMazzola) [reply packages/game-app/src/_shared/index.ts:6]: Isn't it because of the emulators?

File: packages/game-app/src/_shared/index.ts (hunk 4 above)
Comment (rams23) [reply packages/game-app/src/_shared/index.ts:27]: firebase exposes inconsistent api for the useEmulator. You can see the doc

File: packages/game-app/src/signup/apis/executeSignup.ts (hunk 2 above)
Comment (rams23) [reply packages/game-app/src/signup/apis/executeSignup.ts:26]: errors are handled and saved into the store using the saga util addRequestStatusmanagement, and then showed in the ui using the hook. Why do we need custom handling for firebase errors? they are in the form {code, message}. I think that the only thing that is missing is the custom wording using i18n, as underlined in the pr description

File: packages/game-app/src/_shared/index.ts (hunk 1 above)
Comment (rams23) [reply packages/game-app/src/_shared/index.ts:6]: yes excacly, i'll try to understand the impact on the main bundle size and try to find a workaround.

**Comment by rams23** [comment]
lgtm

File: packages/game-app/src/_shared/index.ts (hunk 4 above)
Comment (GabrieleMazzola) [reply packages/game-app/src/_shared/index.ts:27]: lol
